### PR TITLE
Better error handling if rotki server gives unexpected response

### PR DESCRIPTION
**Pull request by LefterisJP**
Partially address what is seen at
https://github.com/rotki/rotki/issues/6282 if the server ever responds with unexpected results again.

### CODE REVIEW

File: rotkehlchen/premium/premium.py [hunk 1]
```diff
@@ -296,12 +296,19 @@ def query_last_data_metadata(self) -> RemoteMetadata:
             raise RemoteError(msg) from e
 
         result = _process_dict_response(response)
```
Comment (yabirgb) [inline rotkehlchen/premium/premium.py:298]: should we check the status code here? The server returned a 404 that is allowed in this function but doesn't raise any error

File: rotkehlchen/premium/premium.py [hunk 2]
```diff
@@ -296,12 +296,19 @@ def query_last_data_metadata(self) -> RemoteMetadata:
...
+        try:
+            metadata = RemoteMetadata(
+                upload_ts=Timestamp(result['upload_ts']),
+                last_modify_ts=Timestamp(result['last_modify_ts']),
+                data_hash=result['data_hash'],
+                data_size=result['data_size'],
+            )
+        except KeyError as e:
+            msg = f'Problem connecting to rotki server. last_data_metadata response missing {e!s} key'  # noqa: E501
+            log.error(f'{msg}. Response was {result}')
```
Comment (yabirgb) [inline rotkehlchen/premium/premium.py:309]: Also here the error is silent for the user. The only thing that will be noticed is that the account doesn't have the premium status

Comment (LefterisJP) [reply rotkehlchen/premium/premium.py:309]: > Also here the error is silent for the user. The only thing that will be noticed is that the account doesn't have the premium status

that's how all these errors are so far handled by rotki no? All failures to connec to rotki server.

File: rotkehlchen/premium/premium.py (hunk 1 above)
Comment (LefterisJP) [reply rotkehlchen/premium/premium.py:298]: ah good point. Will add it.

**Review by yabirgb — COMMENTED** [review]
test need adjustment

**Review by yabirgb — APPROVED** [review]
LGTM

**Pull request merged**